XamarinCommunityToolkit/Helpers/BindablePropertyBuilder.shared.cs

```diff
@@ -0,0 +1,180 @@
+		{
+			bpParameters.BindingPropertyChangedDelegate = propertyChangedDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate that runs before a bound value is changed.
+		/// </summary>
+		/// <param name="propertyChangingDelegate">The delegate that runs before a bound value is changed.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetPropertyChangingDelegate(BindableProperty.BindingPropertyChangingDelegate propertyChangingDelegate)
+		{
+			bpParameters.BindingPropertyChangingDelegate = propertyChangingDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate used to coerce the range of a value.
+		/// </summary>
+		/// <param name="coerceValueDelegate">A delegate used to coerce the range of a value.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetCoerceValueDelegate(BindableProperty.CoerceValueDelegate coerceValueDelegate)
+		{
+			bpParameters.CoerceValueDelegate = coerceValueDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate used to initialize the default value for reference types.
+		/// </summary>
+		/// <param name="createDefaultValueDelegate">The delegate used to initialize the default value for reference types.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetCreateDefaultValueDelegate(BindableProperty.CreateDefaultValueDelegate createDefaultValueDelegate)
+		{
+			bpParameters.CreateDefaultValueDelegate = createDefaultValueDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Instantiates the <see cref="BindableProperty"/> based on the values that have been set.
+		/// </summary>
+		/// <returns>An instance of <see cref="BindableProperty"/> with the values that have been set</returns>
+		/// <exception cref="ArgumentNullException">
+		/// If any of the following are null:
+		/// <see cref="BindableProperty.PropertyName"/>
+		/// <see cref="BindableProperty.ReturnType"/>
+		/// <see cref="BindableProperty.DeclaringType"/>
+		/// </exception>
+		/// <exception cref="ArgumentException">
+		/// If the type of the <see cref="BindableProperty.DefaultValue"/>
+		/// and <see cref="BindableProperty.ReturnType"/> don't match.
+		/// </exception>
+		public BindableProperty Build()
+		{
+			CheckParameter(!string.IsNullOrEmpty(bpParameters.PropertyName), nameof(bpParameters.PropertyName));
+			CheckParameter(bpParameters.ReturnType != default, nameof(bpParameters.ReturnType));
+			CheckParameter(bpParameters.DeclaringType != default, nameof(bpParameters.DeclaringType));
+			if (bpParameters.DefaultValue != null && bpParameters.ReturnType != bpParameters.DefaultValue.GetType())
```

Comment:
This check is wrong
1) DefaultValue can be null (for reference types)
2) DefaultValueCreator can be used instead of DefaultValue (so default value will be null as well)

Reply:
I missed this in my review and agree, I actually just built an attached bindable property the other day where it is common to be null

Reply:
Please remove this check :)